sdk/keyvault/keyvault-keys/src/index.ts

```diff
@@ -85,93 +85,166 @@ export { ProxyOptions, TelemetryOptions, RetryOptions };
  * The client to interact with the KeyVault keys functionality
  */
 export class KeysClient {
-  /**
-   * A static method used to create a new Pipeline object with the provided Credential.
-   *
-   * @static
-   * @param {ServiceClientCredentials | TokenCredential} The credential to use for API requests.
-   * @param {NewPipelineOptions} [pipelineOptions] Optional. Options.
-   * @returns {Pipeline} A new Pipeline object.
-   * @memberof KeysClient
-   */
-  public static getDefaultPipeline(
-    credential: ServiceClientCredentials | TokenCredential,
-    pipelineOptions: NewPipelineOptions = {}
-  ): Pipeline {
-    // Order is important. Closer to the API at the top & closer to the network at the bottom.
-    // The credential's policy factory must appear close to the wire so it can sign any
-    // changes made by other factories (like UniqueRequestIDPolicyFactory)
-    const retryOptions = pipelineOptions.retryOptions || {};
+  private getKeyFromKeyBundle(keyBundle: KeyBundle): Key {
+    const parsedId = parseKeyvaultEntityIdentifier(
+      "keys",
+      keyBundle.key ? keyBundle.key.kid : undefined
+    );
 
-    const userAgentString: string = KeysClient.getUserAgentString(pipelineOptions.telemetry);
+    let resultObject;
+    if (keyBundle.attributes) {
+      resultObject = {
+        keyMaterial: keyBundle.key,
+        ...keyBundle,
+        ...parsedId,
+        ...keyBundle.attributes
+      };
+      delete resultObject.attributes;
+    } else {
+      resultObject = {
+        keyMaterial: keyBundle.key,
+        ...keyBundle,
+        ...parsedId
+      };
+    }
 
-    const requestPolicyFactories: RequestPolicyFactory[] = [
-      proxyPolicy(getDefaultProxySettings((pipelineOptions.proxyOptions || {}).proxySettings)),
-      userAgentPolicy({ value: userAgentString }),
-      generateClientRequestIdPolicy(),
-      deserializationPolicy(), // Default deserializationPolicy is provided by protocol layer
-      throttlingRetryPolicy(),
-      systemErrorRetryPolicy(),
-      exponentialRetryPolicy(
-        retryOptions.retryCount,
-        retryOptions.retryIntervalInMS,
-        RetryConstants.MIN_RETRY_INTERVAL_MS, // Minimum retry interval to prevent frequent retries
-        retryOptions.maxRetryDelayInMs
-      ),
-      redirectPolicy(),
-      isTokenCredential(credential)
-        ? bearerTokenAuthenticationPolicy(credential, "https://vault.azure.net/.default")
-        : signingPolicy(credential)
-    ];
+    return resultObject;
+  }
 
-    return {
-      httpClient: pipelineOptions.HTTPClient,
-      httpPipelineLogger: pipelineOptions.logger,
-      requestPolicyFactories
-    };
+  private getKeyAttributesFromKeyItem(keyItem: KeyItem): KeyAttributes {
```

Comment:
Why are there so many changes here? If it's because private methods are being grouped together, I'd rather not do that as they're currently organized by public methods and private method helpers.

Reply:
Yes, it's exactly because of that. Our project level ESLint rules force us to have certain order of public and private functions :/ not doing it requires us to either have custom ESLint configs or to put the configuration deviation in the header.

Reply:
Can we remove that lint then?

Reply:
you mean disable this order restriction for us? yes we can :) I'll add it at the top of the file